rootless-install.sh

```diff
@@ -0,0 +1,264 @@
+#!/bin/sh
+
+# This script is meant for quick & easy install via:
+#   $ curl -fsSL https://rootless.docker.com -o get-docker.sh
+#   $ sh get-docker.sh
+#
+# NOTE: Make sure to verify the contents of the script
+#       you downloaded matches the contents of install.sh
+#       located at https://github.com/docker/docker-install
+#       before executing.
+#
+# Git commit from https://github.com/docker/docker-install when
+# the script was uploaded (Should only be modified by upload job):
+SCRIPT_COMMIT_SHA=UNKNOWN
+
+# This script should be run with an unprivileged user and install/setup Docker under $HOME/bin/.
+
+set -e
+
+init_vars() {
+	BIN="$HOME/bin"
+	DAEMON=dockerd
+	SYSTEMD=
+	if which systemctl >/dev/null 2>&1; then
+		SYSTEMD=1
+	fi
+}
+
+checks() {
+	# OS verification: Linux only, point osx/win to helpful locations
+	case "$(uname)" in
+	Linux)
+		;;
+	*)
+		>&2 echo "Rootless Docker cannot be installed on $(uname)"; exit 1
+		;;
+	esac
+
+	# User verification: deny running as root (unless forced?)
+	if [ "$(id -u)" = "0" ]; then
+		>&2 echo "Refusing to install rootless Docker as the root user"; exit 1
+	fi
+
+	# HOME verification
+	if [ ! -d "$HOME" ]; then
+		>&2 echo "Aborting because HOME directory $HOME does not exist"; exit 1
+	fi
+
+	if [ -d "$BIN" ]; then
+		if [ ! -w "$BIN" ]; then
+			>&2 echo "Aborting because $BIN is not writable"; exit 1
+		fi
+	else
+		if [ ! -w "$HOME" ]; then
+			>&2 echo "Aborting because HOME (\"$HOME\") is not writable"; exit 1
+		fi
+	fi
+
+	# Existing rootful docker verification
+	if [ -w /var/run/docker.sock ]; then
+		>&2 echo "Aborting because rootful Docker is running and accessible"; exit 1
+	fi
+
+
+	if [ ! -w "$XDG_RUNTIME_DIR" ]; then
+		if [ -n "$SYSTEMD" ]; then
+			>&2 echo "Aborting because systemd was detected but XDG_RUNTIME_DIR (\"$XDG_RUNTIME_DIR\") does not exist or is not writable"
+			>&2 echo "Hint: this could happen if you changed users with 'su' or 'sudo'. To work around this:"
+			>&2 echo "- try again by first running with root privileges 'loginctl enable-linger <user>' where <user> is the unprivileged user and export XDG_RUNTIME_DIR to the value of RuntimePath as shown by 'loginctl show-user <user>'"
+			>&2 echo "- or simply log back in as the desired unprivileged user (ssh works for remote machines)"
+			exit 1
+		fi
+		export XDG_RUNTIME_DIR="/tmp/docker-rootless-$(id -u)"
```

Comment:
Does it work on non-bash sh?

Reply:
@AkihiroSuda which part shouldn't work without bash? The entire file is run with `/bin/sh`

Reply:
Ancient sh required `export FOO=BAR` to be written as `FOO=BAR; export FOO`.

But recent implementation such as `dash` and `ash` seem to support `export FOO=BAR`...